src/Microsoft.VisualStudio.LanguageServices.Razor/IRazorEditorWorkerProvider.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.VisualStudio.Text;
+
+namespace Microsoft.VisualStudio.LanguageServices.Razor
+{
+    public interface IRazorEditorWorkerProvider
```

Comment:
Let's make these abstract classes and call em `...Factories`.

In general the VS convention is to use works like `FooServiceFactory` and `FooService`. I think `RazorEditorWorker` is a good name.